features/support/env.rb

```diff
@@ -1,84 +1,41 @@
-# Any 'run once' setup should go here as this file is evaluated
-# when the environment loads.
-# Any helper functions added here will be available in step
-# definitions
-
-RUNNING_CI = ENV['TRAVIS'] == 'true'
-
-# Max time in seconds to wait for an action to complete
-MAX_WAIT_TIME = RUNNING_CI ? 300 : 60
-
-ENV['MAZE_SDK'] = '12.1' unless ENV['MAZE_SDK']
-MAZE_SDK = ENV['MAZE_SDK']
-
-Dir.chdir('features/fixtures/ios-swift-cocoapods') do
-  run_required_commands([
-    ['bundle', 'install'],
-    ['bundle', 'exec', 'pod', 'install'],
-    ['../../scripts/build_ios_app.sh'],
-    ['../../scripts/remove_installed_simulators.sh'],
-    ['../../scripts/launch_ios_simulators.sh'],
-  ])
-end
-
-
-# Scenario hooks
-Before do
-  # Name set in launch_ios_simulators.sh
-  set_script_env('iOS_Simulator', 'maze-sim')
-end
-
-After do
-  # Clean up caches
-  files = Dir.glob("#{app_file_path}/**/Library/Caches/{KSCrashReports/,bugsnag_}*")
-  files.each {|f| FileUtils.rm_rf(f) }
-end
-
-at_exit do
-  run_required_commands([
-    ['features/scripts/remove_installed_simulators.sh'],
-  ])
-end
-
-def app_file_path
-  app_path = `xcrun simctl get_app_container maze-sim com.bugsnag.iOSTestApp`.chomp
-  app_path.gsub(/(.*Containers).*/, '\1')
-end
-
-def test_app_pid
-  output = `xcrun simctl spawn maze-sim launchctl print system | grep UIKitApplication:com.bugsnag.iOSTestApp`
-  pattern = /(\d+)\s+(-|-?\d+?)\s+UIKitApplication:com.bugsnag.iOSTestApp/
-  match = output.match(pattern)
-  if match.nil?
-    nil
-  else
-    match[1]
+# Configure app environment
+bs_username = ENV['BROWSER_STACK_USERNAME']
+bs_access_key = ENV['BROWSER_STACK_ACCESS_KEY']
+bs_local_id = ENV['BROWSER_STACK_LOCAL_IDENTIFIER'] || 'maze_browser_stack_test_id'
+bs_device = ENV['DEVICE_TYPE']
+app_location = ENV['APP_LOCATION']
+
+# Set this explicitly
+$api_key = "12312312312312312312312312312312"
+
+
+After do |scenario|
+  if $driver
+    # [:syslog, :crashlog, :performance, :server, :safariConsole, :safariNetwork]
+    # puts $driver.driver.logs.get(:crashlog)
+    $driver.reset
   end
 end
 
-def test_app_is_running?
-  pid = test_app_pid
-  !pid.nil? # check that PID is valid
+AfterConfiguration do |config|
+  AppAutomateDriver.new(
+    bs_username,
+    bs_access_key,
+    bs_local_id,
+    bs_device,
+    app_location,
+    :accessibility_id,
+    {
+      'browserstack.deviceLogs' => true,
+      'browserstack.appium_version' => '1.16.0'
```

Comment:
Interesting, hadn't seen that we set the version here.  I don't think the iOS 10 phone we use likes it for some reason despite BrowserStack website saying it is supported.  Possibly a case for contacting BrowserStack support.
Sorry, this isn't really a review comment tbh - no action needed here as such.